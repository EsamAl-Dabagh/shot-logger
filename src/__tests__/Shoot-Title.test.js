import React, { Component } from 'react';
import { shallow } from 'enzyme';
import ShootTitle from '../components/Shoot-Title';

describe("Shoot Title", () => {
  const shootTitle = shallow(<ShootTitle />);

  it("renders correctly", () => {
    expect(shootTitle).toMatchSnapshot();
  });

  it("initializes with empty text string", () => {
    expect(shootTitle.state()).toEqual({ text: "" });
  });

  describe("when `text` is empty", () => {
    it("displays input box", () => {
      expect(shootTitle.find(".title-input").exists()).toBe(true);
    });
  });

  describe("when `text` is not empty", () => {
    beforeEach(() => {
      shootTitle.setState({ text: "Day 01" })
    });

    it("displays the title", () => {
      expect(shootTitle.find(".shoot-title").exists()).toBe(true);
    });
  });
});