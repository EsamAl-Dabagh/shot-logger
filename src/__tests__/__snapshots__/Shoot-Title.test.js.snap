// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shoot Title renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShootTitle />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <form>
        <TextField
          className="title-input"
          onKeyDown={[Function]}
          placeholder="Enter Shoot Title"
          required={false}
          select={false}
          variant="standard"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <TextField
          className="title-input"
          onKeyDown={[Function]}
          placeholder="Enter Shoot Title"
          required={false}
          select={false}
          variant="standard"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "title-input",
          "onKeyDown": [Function],
          "placeholder": "Enter Shoot Title",
          "required": false,
          "select": false,
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "form",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <form>
          <TextField
            className="title-input"
            onKeyDown={[Function]}
            placeholder="Enter Shoot Title"
            required={false}
            select={false}
            variant="standard"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <TextField
            className="title-input"
            onKeyDown={[Function]}
            placeholder="Enter Shoot Title"
            required={false}
            select={false}
            variant="standard"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "title-input",
            "onKeyDown": [Function],
            "placeholder": "Enter Shoot Title",
            "required": false,
            "select": false,
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "form",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
